articles/dms/howto-mysql-to-azure-mysql-powershell.md

```diff
@@ -0,0 +1,657 @@
+
+As part of configuring the migration task, you will create a mapping between the source and target tables. The mapping is at the table name level but the assumption is that the table structure (column count, column names, data types etc.) of the mapped tables is exactly the same.
+
+The following script creates a mapping based on the target and source table list extracted in **Step 7**. For partial data load, the user can provide a list of table to filter out the tables. If no user input is provided, then all target tables are mapped. The script also checks if a table with the same name exists in the source or not. If table name does not exists in the source, then the target table is ignored for migration. 
+
+```powershell
+# Create the source to target table map
+# Optional table settings
+# DEFAULT: $SelectedTables = $null => migrate all tables
+# Array of qualified target table names which should be migrated 
+[string[]] $SelectedTables = @("migtargetdb.coupons", "migtargetdb.daily_cash_sheets");
+
+LogMessage -Message "Creating the table map based on the user input and database table information ..." `
+    -IsProcessing $true
+
+$targetTables = $getTargetTablesTask.Output.DatabasesToTables."$TargetDatabaseName";
+$sourceTables = $getSourceTablesTask.Output.DatabasesToTables."$SourceDatabaseName";
+$tableMap = New-Object 'system.collections.generic.dictionary[string,string]';
+
+$schemaPrefixLength = $($TargetDatabaseName + ".").Length;
+foreach ($tgtTable in $targetTables) {
+    # Removing the database name prefix from the table name so that comparison
+    # can be done in cases where database name given are different
+    $tableName = $tgtTable.Name.Substring($schemaPrefixLength, `
+            $tgtTable.Name.Length - $schemaPrefixLength)
+
+    # Check if the table exists in the source
+    $sourceExists = $false
+    foreach ($sourceTable in $sourceTables) {
+        if ($sourceTable -ieq "$($SourceDatabaseName).$tableName") {
+            $sourceExists = $true
+            break;
+        }
+    }
+
+    # If source does not exists then ignore the table and continue
+    if (-not $sourceExists) { continue; }
+
+    # Check if only a subset of tables on the target should be migrated
+    if ($null -ne $SelectedTables) {
+        $isSelected = $false;
+
+        # Check if this table is on the selected list
+        foreach ($nextSelectedTable in $SelectedTables) {
+            $isSelected = $tgtTable.Name -ieq $nextSelectedTable;
+
+            if ($isSelected) {
+                break;
+            }
+        }
+
+        if (-not $isSelected) {
+            continue;
+        }
+    }
+
+    $tableMap.Add("$($SourceDatabaseName).$tableName", "$($TargetDatabaseName).$tableName");
+}
+
+if ($tableMap.Count -le 0) { throw "ERROR: Could not create table mapping." }
+LogMessage -Message "Migration table mapping created for $($tableMap.Count) tables."
+```
+
+## Step 9: Create and configure the migration task
```

Comment:
Step 9 doesn't actually create the migration task, it just creates the input for the task.

Reply:
Done